File: backend/src/main/java/com/adhamamr/passwordy/dto/PasswordResponse.java
package com.adhamamr.passwordy.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
public class PasswordResponse {
    private Long id;
    private String label;
    private String value;  // encrypted/masked in real implementation remember adham please!!!!
    private String username;
    private String url;
    private String notes;
    private String category;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

}

File: frontend/app/src/main/java/com/adhamamr/passwordy/data/model/PasswordResponse.kt
package com.adhamamr.passwordy.data.model

data class PasswordResponse(
    val id: Long,
    val label: String,
    val value: String,
    val username: String?,
    val url: String?,
    val notes: String?,
    val category: String?,
    val createdAt: String,
    val updatedAt: String
)